pkg/abstractions/common/instance.go

```diff
@@ -287,3 +292,47 @@ func (i *AutoscaledInstance) State() (*AutoscaledInstanceState, error) {
 	state.FailedContainers = failedContainers
 	return &state, nil
 }
+
+func (i *AutoscaledInstance) handleUnhealthyDeploymentEvents(failedContainers []string) {
```

Comment:
since this is also sending events like i.HandleDeploymentHealthy(i.Stub.ExternalId), should it just be called handledeploymentevents?

Reply:
yea that makes sense